azurerm/resource_arm_hdinsight.go

```diff
@@ -0,0 +1,1151 @@
+					gateway[dotKey] = v
+				}
+				configurations["gateway"] = &gateway
+			}
+			rserverInterfaceList := configurationsFlatMap["rserver"].([]interface{})
+			rserverFlatMap, err := expandList(rserverInterfaceList)
+			if err == nil {
+				rserver := make(map[string]interface{})
+				if v := rserverFlatMap["rstudio"].(bool); v != false {
+					rserver["rstudio"] = v
+				}
+				configurations["rstudio"] = &rserver
+			}
+			clusterDefinition.Configurations = &configurations
+		}
+	}
+
+	return clusterDefinition
+}
+
+func expandAzureRmHDInsightSecurityProfile(d *schema.ResourceData) *hdinsight.SecurityProfile {
+	securityProfileInterfaceList := d.Get("security_profile").([]interface{})
+	securityProfile := &hdinsight.SecurityProfile{}
+
+	securityProfileFlatMap, err := expandList(securityProfileInterfaceList)
+	if err != nil {
+		return nil
+	}
+
+	if v := securityProfileFlatMap["directory_type"].(hdinsight.DirectoryType); v != "" {
+		securityProfile.DirectoryType = v
+	}
+
+	if v := securityProfileFlatMap["domain"].(string); v != "" {
+		securityProfile.Domain = &v
+	}
+
+	if v := securityProfileFlatMap["organizational_unit_dn"].(string); v != "" {
+		securityProfile.OrganizationalUnitDN = &v
+	}
+
+	ldapsUrls := []string{}
+	ldapsUrlsList := securityProfileFlatMap["ldaps_urls"].([]interface{})
+	for _, ldapsUrl := range ldapsUrlsList {
+		if v := ldapsUrl.(string); v != "" {
+			ldapsUrls = append(ldapsUrls, v)
+		}
+	}
+	securityProfile.LdapsUrls = &ldapsUrls
+
+	if v := securityProfileFlatMap["domain_username"].(string); v != "" {
+		securityProfile.DomainUsername = &v
+	}
+
+	if v := securityProfileFlatMap["domain_password"].(string); v != "" {
+		securityProfile.DomainUserPassword = &v
+	}
+
+	clusterUsersGroupDNS := []string{}
+	clusterUsersGroupDNSList := securityProfileFlatMap["cluster_users_group_dns"].([]interface{})
+	for _, clustUsersGroupDNSItem := range clusterUsersGroupDNSList {
+		if v := clustUsersGroupDNSItem.(string); v != "" {
+			clusterUsersGroupDNS = append(clusterUsersGroupDNS, v)
+		}
+	}
+	securityProfile.ClusterUsersGroupDNS = &clusterUsersGroupDNS
+
+	return securityProfile
+}
+
+func expandAzureRmHDInsightStorageProfile(d *schema.ResourceData) *hdinsight.StorageProfile {
+	storageProfileInterfaceList := d.Get("storage_profile").([]interface{})
+	storageProfile := &hdinsight.StorageProfile{}
+
+	storageProfileFlatMap, err := expandList(storageProfileInterfaceList)
+	if err != nil {
+		return nil
+	}
+
+	storageAccounts := []hdinsight.StorageAccount{}
+	storageAccountsInterfaceList := storageProfileFlatMap["storage_accounts"].([]interface{})
+
+	for _, storageAccountInterface := range storageAccountsInterfaceList {
+		storageAccountFlatMap := storageAccountInterface.(map[string]interface{})
+		var storageAccount hdinsight.StorageAccount
+		if v := storageAccountFlatMap["name"].(string); v != "" {
+			endpoint := strings.Replace(strings.Replace(v, "https://", "", 1), "/", "", 1)
```

Comment:
so this might make more sense to parse the URL and then string replace the host header?